app-backend/api/src/main/scala/campaign/Routes.scala

```diff
@@ -0,0 +1,250 @@
+        }
+      }
+    }
+  }
+
+  def getCampaign(campaignId: UUID): Route = authenticate { user =>
+    authorizeScope(
+      ScopedAction(Domain.Campaigns, Action.Read, None),
+      user
+    ) {
+      authorizeAuthResultAsync {
+        CampaignDao
+          .authorized(
+            user,
+            ObjectType.Campaign,
+            campaignId,
+            ActionType.View
+          )
+          .transact(xa)
+          .unsafeToFuture
+      } {
+        rejectEmptyResponse {
+          complete {
+            CampaignDao
+              .getCampaignById(campaignId)
+              .transact(xa)
+              .unsafeToFuture
+          }
+        }
+      }
+    }
+  }
+
+  def updateCampaign(campaignId: UUID): Route = authenticate { user =>
+    authorizeScope(
+      ScopedAction(Domain.Campaigns, Action.Update, None),
+      user
+    ) {
+      authorizeAuthResultAsync {
+        CampaignDao
+          .authorized(
+            user,
+            ObjectType.Campaign,
+            campaignId,
+            ActionType.Edit
+          )
+          .transact(xa)
+          .unsafeToFuture
+      } {
+        entity(as[Campaign]) { updatedCampaign =>
+          onSuccess(
+            CampaignDao
+              .updateCampaign(
+                updatedCampaign,
+                campaignId
+              )
+              .transact(xa)
+              .unsafeToFuture
+          ) {
+            completeSingleOrNotFound
+          }
+        }
+      }
+    }
+  }
+
+  def deleteCampaign(campaignId: UUID): Route = authenticate { user =>
+    authorizeScope(
+      ScopedAction(Domain.Campaigns, Action.Delete, None),
+      user
+    ) {
+      authorizeAuthResultAsync {
+        CampaignDao
+          .authorized(
+            user,
+            ObjectType.Campaign,
+            campaignId,
+            ActionType.Delete
+          )
+          .transact(xa)
+          .unsafeToFuture
+      } {
+        onSuccess(
+          CampaignDao
+            .deleteCampaign(campaignId, user)
+            .transact(xa)
+            .unsafeToFuture
+        ) {
+          completeSingleOrNotFound
+        }
+      }
+    }
+  }
+
+  def listCampaignUserActions(campaignId: UUID): Route = authenticate { user =>
+    authorizeScope(
+      ScopedAction(Domain.Campaigns, Action.ReadPermissions, None),
+      user
+    ) {
+      authorizeAuthResultAsync {
+        CampaignDao
+          .authorized(
+            user,
+            ObjectType.Campaign,
+            campaignId,
+            ActionType.View
+          )
+          .transact(xa)
+          .unsafeToFuture
+      } {
+        user.isSuperuser match {
+          case true => complete(List("*"))
+          case false =>
+            onSuccess(
+              CampaignDao
+                .getCampaignById(campaignId)
+                .transact(xa)
+                .unsafeToFuture
+            ) { campaignO =>
+              complete {
+                (campaignO map { campaign =>
```

Comment:
This could be a `traverse` instead of an `Option[T] map { ... } getOrElse { foo.pure[F] }`, but you can take or leave this